File: leetcode/lc-us/cpp/997.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int findJudge(int n, vector<vector<int>> &trust) {
        int people_who_are_trusted[1001] = {};
        int people_who_trust_others[1001] = {};
        for (const vector<int> &t: trust) {
            people_who_are_trusted[t[1]]++;
            people_who_trust_others[t[0]]++;
        }
        for (int i = 1; i <= 1000; i++) {
            if (people_who_are_trusted[i] == n - 1) {
                if (!people_who_trust_others[i]) {
                    return i;
                }
            }
        }
        return -1;
    }
};

File: leetcode/lc-us/py/997.py
from typing import List


class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        in_degree, out_degree = [0] * (n + 1), [0] * (n + 1)
        for t in trust:
            in_degree[t[1]] += 1
            out_degree[t[0]] += 1
        for i in range(1, n + 1):
            if in_degree[i] == n - 1 and out_degree[i] == 0:
                return i
        return -1
